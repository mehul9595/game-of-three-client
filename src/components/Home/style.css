.game-home {
    width: 100%;
    height: 100%;
}

.game-home .ant-btn.ant-btn-primary.game-button {
  height: 4em;
  font-size: 16px;
  width: 13em;
  margin: 10px;
  padding-left: 20px;
  border-radius: 50px;
  float: left;
  color: #fff;
  box-shadow: 5px 5px 2px -2px #a5a5a5;
}
