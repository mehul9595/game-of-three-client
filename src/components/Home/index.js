import React from "react";
import { Card, Button, Divider, Row, Col } from "antd";
import { PlayCircleOutlined } from "@ant-design/icons";
import "./style.css";
import "antd/dist/antd.css";

const Home = (props) => {
  return (
    <Card bordered hoverable className="game-home">
      <h1>Game of Three @ Scoober</h1>
      <Divider className="game-home-divider" />
      <Row justify="center">
        <Col>
          <Button
            type="primary"
            icon={<PlayCircleOutlined />}
            htmlType="button"
            className="game-button"
            onClick={() => props.history.push("/singleplayer")}
          >
            Single Player
          </Button>
        </Col>
      </Row>
      <Row justify="center">
        <Col>
          <Button
            type="primary"
            htmlType="button"
            className="game-button"
            icon={<PlayCircleOutlined />}
            onClick={() => props.history.push("/multiplayer")}
            data-testId="multiplayerBtn"
          > 
            Multiplayer
          </Button>
        </Col>
      </Row>
    </Card>
  );
};

export default Home;
