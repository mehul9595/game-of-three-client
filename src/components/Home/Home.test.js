import React from "react";
import renderer from 'react-test-renderer';
import {  cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Home from "./index";

describe("HomeComponent", () => {
  afterEach(cleanup);
  const wrapper = shallow(<Home />);
  const gameName = "Game of Three @ Scoober";

  it("should match the header", () => {
    expect(wrapper.find("h1").contains(gameName)).toBe(true);
  });

  it("should contain 2 buttons", () => {
    expect(wrapper.find("Button")).toHaveLength(2);
  });

  test('should contain Singleplayer button', () => {
    expect(wrapper.find("Button").contains("Single Player")).toBe(true);

  });

  test('should contain multiplayer button', () => {
    expect(wrapper.find({ 'data-testId': 'multiplayerBtn' })).toHaveLength(1);
  });
  
  

  it("should render sucessfully with snapshot", () => {
    
     const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
