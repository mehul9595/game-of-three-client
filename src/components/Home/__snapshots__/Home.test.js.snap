// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render sucessfully with snapshot 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable game-home"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <h1>
      Game of Three @ Scoober
    </h1>
    <div
      className="ant-divider ant-divider-horizontal game-home-divider"
      role="separator"
    />
    <div
      className="ant-row ant-row-center"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <button
          className="ant-btn ant-btn-primary game-button"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="play-circle"
            className="anticon anticon-play-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
              />
            </svg>
          </span>
          <span>
            Single Player
          </span>
        </button>
      </div>
    </div>
    <div
      className="ant-row ant-row-center"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <button
          className="ant-btn ant-btn-primary game-button"
          data-testid="multiplayerBtn"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="play-circle"
            className="anticon anticon-play-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
              />
            </svg>
          </span>
          <span>
            Multiplayer
          </span>
        </button>
      </div>
    </div>
    <div
      className="ant-row ant-row-center"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <button
          className="ant-btn ant-btn-primary game-button"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="play-circle"
            className="anticon anticon-play-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
              />
            </svg>
          </span>
          <span>
            Single Player Redux
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
